src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForExpressionProfileBuildingBlock.cs

```diff
@@ -28,6 +36,35 @@ public IReadOnlyList<ExpressionProfileBuildingBlock> LoadFromPKML()
          return (string.IsNullOrEmpty(filename) ? Enumerable.Empty<ExpressionProfileBuildingBlock>() : LoadItems(filename)).ToList();
       }
 
+      public IMoBiCommand UpdateExpressionProfileFromDatabase(ExpressionProfileBuildingBlock buildingBlock)
+      {
+         var expressionProfileUpdate = _pkSimStarter.UpdateExpressionProfileFromDatabase(buildingBlock);
+         
+         if (expressionProfileUpdate == null)
+            return new MoBiEmptyCommand();
+
+         var macroCommand = new MoBiMacroCommand
+         {
+            ObjectType = ObjectTypes.ExpressionProfileBuildingBlock,
+            CommandType = AppConstants.Commands.EditCommand,
+            Description = AppConstants.Commands.UpdateRelativeExpressions
+         };
+
+         macroCommand.AddRange(expressionProfileUpdate.Select(parameter => updateCommandFor(buildingBlock, parameter.Path, parameter.UpdatedValue)));
+
+         return macroCommand.Run(Context);
+      }
+
+      private static ICommand updateCommandFor(ExpressionProfileBuildingBlock buildingBlock, ObjectPath path, double? value)
+      {
+         var parameterToUpdate = buildingBlock.FirstOrDefault(x => Equals(x.Path, path));
+         
+         if (parameterToUpdate == null)
```

Comment:
can you verify in the history that you do not get empty command int he parent macro command?

Reply:
Yes, I verified empty commands don't appear in the history.